### DISCUSSION TITLE
Creating functionality for loading and storing model configurations and parameters

### DISCUSSION THREAD

**Discussion by jrego43** [LazaroHurtado/micrograd-rs#12] [Ideas]
It is imperative that every deep learning framework allows for the loading and storing of trained neural networks so that they can actually be put to use in software applications.

Here we should discuss how we will go about implementing loading and storing systems that are fully compatible. 

I believe the following webpages provide useful information as they show how to load and store models in the
- [PyTorch way](https://pytorch.org/tutorials/beginner/saving_loading_models)
- [TensorFlow way](https://www.tensorflow.org/tutorials/keras/save_and_load)

**Comment by jrego43** [comment]
Would this warrant a new module @LazaroHurtado ? Something like io?

**Reply by LazaroHurtado** [reply]
In terms of saving or loading parameters I believe not. I believe we can get away with just creating a new method in the `Module` trait, this would leverage the existing `parameters` method and use it to either save to a pickle full or loading from.
But I'm curious why you are proposing a new module, is there something I'm overlooking?

**Reply by jrego43** [reply]
I wasn't really proposing, just unfamiliar with the general structure of such frameworks and wanted the clarification.
